typescript/core/src/IWritable.ts

```diff
@@ -7,4 +7,7 @@ export interface IWritable {
 
   // The current position in bytes from the start of the output
   position(): bigint;
+
+  // Seek the cursor to the given position
+  seek(position: bigint): Promise<void>;
```

Comment:
I think most of our existing writers could be updated to add seeking support, but I think it would be better to scope this requirement down so it's only needed when opening in append mode. As-is this would also be a breaking change which would require a major version bump.